plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java

```diff
@@ -88,6 +90,23 @@ public Connection(IClient client, ICredentialsPrompter credentialsPrompter, ISSL
 		this.sslCertificateCallback = sslCertCallback;
 	}
 	
+	@Override
+	public Properties getExtendedProperties() {
+		return extendedProperties;
+	}
+
+	@Override
+	public void setExtendedProperty(String name, String value) {
+		Properties oldExt = new Properties(extendedProperties);
```

Comment:
I always found default properties to be tricky. For instance, considering:
```java
  Properties p1 = new Properties();
  p1.put("foo", "bar");
  Properties p2 = new Properties(p1);
```
then p1.equals(p2) return false!